apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,60 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
+  String variant
+
+  @InputFiles
+  Collection<File> getInputFiles() {
+    Set<File> inputFiles = Sets.newHashSet()
+    config.each { ext ->
+      ext.getFiles().asMap().entrySet().each { entry ->
+        entry.value.each { file ->
+          inputFiles.add(new File(entry.key, file))
+        }
+      }
+    }
+    return inputFiles
+  }
+
+  @Override
+  public void exec() {
```

Comment:
Thanks for the thorough explanation @marwanad.
Bullet point 1 -> I think that's exactly what we want (one IR for all the graphql files)

```
I think a good solution to keep the flow neat and maintain the package structure would be to 
have the relative path as a field in the IR.
```
^ You mean because you need to be able to generate the java interfaces in the same directory hierarchy as the original .graphql files?

Bullet point 3, item i -> 👍 , do we need to check for that though, or just let apollo-codegen fail for us?
Bullet point 3, item ii -> Sounds like we should follow the same lookup pattern that android uses. First in the variant specific directory (`src/buildTypeFlavorName/graphql`), then build type (`src/buildType/graphql`) and finally main (`src/main/graphql`)